src/Servers/Kestrel/perf/Kestrel.Performance/Mocks/MockHttpContextFactory.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Performance
+{
+    public class MockHttpContextFactory : IHttpContextFactory
+    {
+        private readonly object _lock = new object();
+        private readonly List<DefaultHttpContext> _cache = new List<DefaultHttpContext>();
```

Comment:
Queue?

Reply:
I tried `ConcurrentQueue<T>` and noticed it allocates a node. Does a regular queue allocate?

Reply:
Regular queue is backed by an array, it just makes the code more clear.